### ISSUE TITLE
Outlook tools not being Recognized in 2018.1?

### ISSUE THREAD

**Issue by pgensler** [rpaugh/Alteryx-Outlook-Tools#5]
Hi @rpaugh  I recently tried to install the Outlook tools from the link provided, and I can't seem to get 2018.1.4.44311 to recognize the tools. 

It might be worth noting in the readme how to manually install the tool by placing the `.dll` files into the correct location for it work properly. Is the tool compatible with 2018.1?

**Comment by rpaugh** [comment]
Hi @pgensler, I'm actually not sure if the tool is compatible with 2018.1 as I don't yet have that version.  That said, I don't believe I'm using any programming that would be affected by an Alteryx version change.  I will see if there's a way for me to upgrade to 2018 and test, but in the meantime here is some information that may help:

It's possible that the DLLs installed to the correct location, but the .ini file that tells Alteryx where to look for those DLLs may not have the right path.  Locate your OutlookInputTool.ini file on the server, open it in notepad, and make sure the path in it points to the location you selected during installation.  

For example, in my case the DLLs installed to:
C:\Program Files\Alteryx\Outlook

...my OutlookInputTool.ini installed to:
C:\Program Files\Alteryx\Settings\AdditionalPlugins

...and the x74path in the .ini file is:
C:\Program Files\Alteryx\Outlook

Another thing to keep in mind is that the directory in which you installed the tool needs to be accessible by Alteryx, which is why the default location is the same as the Alteryx Designer's default directory.  

If you're dealing with a desktop installation, verify that either the tool or Alteryx itself was not installed for a particular user (i.e. in the user's AppData folder).  If it is, then simply locate the .ini file mentioned above and change the x64 path in the file to match the Alteryx installation.

**Comment by pgensler** [comment]
HI @rpaugh It looks like it did install properly but it just seemed to blend in with the other connectors,  as it looks odd that I can't see the whole name of the tool on two separate lines like with SalesforceInput. Maybe a simple line break character is needed?

Also, it would definitely help to include a brief description so when you hover over it, you can see what the tool is intended to do.
![image](https://user-images.githubusercontent.com/7308115/39059318-c8eca566-4483-11e8-8952-d26b0310954e.png)

Let me see if I can find a link on how to do this, as I've tried to look at this in the past for a connector I tried to build.

**Comment by rpaugh** [comment]
From what I could tell, the name of the tool was populated from the class name reference in the code:
``public class OutlookInputTool : AlteryxGuiToolkit.Plugins.IPlugin
{
}``
I noticed this issue right away as well and tried changing the name of the DLL and the entry point reference, but it didn't work.  There still seems to be a few quirks like this with the C# SDK, and based on a previous conversation I had during development I got the impression that not much attention has been given to the C# SDK in recent years due to the focus on HTML/JS and Python SDKs.  For example, there's no way to tell your tool point to a custom page when you click the Help icon in the UI.  You have to create a custom help link within your control's UI instead.  Things like that.

I didn't see anywhere to put in a description either, so if you know how to do that, I'd be happy to update it.

**Comment by pgensler** [comment]
So, I'm not familiar with the C# SDK, but from here: https://github.com/jdunkerley/AlteryxAddIns/blob/master/OmnibusRegex/OmnibusRegExConfig.xml
it looks like it is possible within the xml to edit it. Maybe reaching out to support to see if you can do this with the C# SDK might be a conversation to have? I think Jame's tools have the same feature, so I don't think it's out of reach to try.

**Comment by rpaugh** [comment]
It looks like those values are only available in the Javascript SDK:

![image](https://user-images.githubusercontent.com/30471541/39131481-7ec8cc56-46dd-11e8-916b-db2a0bc53dd2.png)

I reviewed his XmlInput C# tool as well and didn't see anything about setting the tool name or description.

**Comment by pgensler** [comment]
Ah, yeah that might be something to post in the Dev Space or feedback:
https://community.alteryx.com/t5/Dev-Space/bd-p/developer-discussions
https://community.alteryx.com/t5/Dev-Space/SURVEY-Alteryx-Developer-Community-Feedback/m-p/87570

**Comment by rpaugh** [comment]
I think this is just a limitation with the C# SDK that Alteryx is not really trying to fix.  They're really trying to push people to the HTML GUI SDK.  I found a post where someone asked the same question and the response was to use that SDK instead.  When I have time I'll probably rebuild the GUI with the new HTML SDK and have it call my existing C# DLL.  Then I'll be able to set the tool name, description, example, etc.  Hopefully I'll be able to get to this soon, but I can't guarantee anything at the moment.  I'll let you know if/when this version of the tool becomes available.

**Comment by CBSJeff** [comment]
Just installed, but it does not appear in Connector group. Also, this post refers to OutlookInputTool.ini. It did not install OutlookInputTool.ini but OutlookInputPlugin.ini. Is that correct? I'm trying to install on 11.8.

**Comment by rpaugh** [comment]
@CBSJeff , yes, OutlookInputPlugin.ini is correct.  Check out the post above regarding location of Alteryx DLLs and the OutlookInputTool DLLs, and make sure that both are in the correct location and that the path in the .ini file is pointing to the correct location.
